fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -541,10 +519,17 @@ class DefaultTabsTrayController(
     }
 
     override fun handleTabSelected(tab: TabSessionState, source: String?) {
-        TabsTray.openedExistingTab.record(TabsTray.OpenedExistingTabExtra(source ?: "unknown"))
-        tabsUseCases.selectTab(tab.id)
-        browsingModeManager.mode = BrowsingMode.fromBoolean(tab.content.private)
-        handleNavigateToBrowser()
+        val selected = tabsTrayStore.state.mode.selectedTabs
```

Comment:
It's a bit sneaky, but could we update the name of this function to use `Clicked` instead of `Selected` while we're here? I think that lines up a bit better with the rest of the methods, and I think it'll make the path more clear when we refactor this to an action.

Reply:
I'm on the same wavelength as you. This interface function is from an [AC interface](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsTray.kt), and I actually have a [ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1814243) to decouple us from that and port the consumers to the interactor.